homeassistant/components/romy/vacuum.py

```diff
@@ -0,0 +1,115 @@
+"""Support for Wi-Fi enabled ROMY vacuum cleaner robots.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/vacuum.romy/.
+"""
+
+
+from typing import Any
+
+from romy import RomyRobot
+
+from homeassistant.components.vacuum import StateVacuumEntity, VacuumEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, LOGGER
+from .coordinator import RomyVacuumCoordinator
+
+ICON = "mdi:robot-vacuum"
+
+FAN_SPEED_NONE = "default"
+FAN_SPEED_NORMAL = "normal"
+FAN_SPEED_SILENT = "silent"
+FAN_SPEED_INTENSIVE = "intensive"
+FAN_SPEED_SUPER_SILENT = "super_silent"
+FAN_SPEED_HIGH = "high"
+FAN_SPEED_AUTO = "auto"
+
+FAN_SPEEDS: list[str] = [
+    FAN_SPEED_NONE,
+    FAN_SPEED_NORMAL,
+    FAN_SPEED_SILENT,
+    FAN_SPEED_INTENSIVE,
+    FAN_SPEED_SUPER_SILENT,
+    FAN_SPEED_HIGH,
+    FAN_SPEED_AUTO,
+]
+
+# Commonly supported features
+SUPPORT_ROMY_ROBOT = (
+    VacuumEntityFeature.BATTERY
+    | VacuumEntityFeature.RETURN_HOME
+    | VacuumEntityFeature.STATE
+    | VacuumEntityFeature.START
+    | VacuumEntityFeature.STOP
+    | VacuumEntityFeature.FAN_SPEED
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up ROMY vacuum cleaner."""
+
+    coordinator: RomyVacuumCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([RomyVacuumEntity(coordinator, coordinator.romy)], True)
+
+
+class RomyVacuumEntity(CoordinatorEntity[RomyVacuumCoordinator], StateVacuumEntity):
+    """Representation of a ROMY vacuum cleaner robot."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_supported_features = SUPPORT_ROMY_ROBOT
+    _attr_fan_speed_list = FAN_SPEEDS
+    _attr_icon = ICON
+
+    def __init__(
+        self,
+        coordinator: RomyVacuumCoordinator,
+        romy: RomyRobot,
+    ) -> None:
+        """Initialize the ROMY Robot."""
+        super().__init__(coordinator)
+        self.romy = romy
+        self._attr_unique_id = self.romy.unique_id
+        self._device_info = DeviceInfo(
```

Comment:
ok. Be aware  the user can change the name of the robot (for example via the normal iot app).
Will a corner case anyhow

Where is this entity name used exactly in home assistant? didn't see any differnce

Reply:
https://developers.home-assistant.io/docs/core/entity/#entity-naming describes how naming works.

Reply:
>  ...and should not include the name of the device or the type of the entity.

Thanks, i am not sure if i understand it correctly, had always wrong assumption that it is the device name (like romy.name returns)


romy.name would be only correct if robot comes direct from factory as soon user sets the name it will not be correct anymore, don't know name how to get this information via the api, so i let it for now as it was:

```
_attr_has_entity_name = True
_attr_name = None
```

Reply:
okay romy.name is product name (assumed wrong as user name)
so changed to your suggestion using romy.name